lineman/app/components/group_form/group_form.coffee

```diff
@@ -0,0 +1,65 @@
+angular.module('loomioApp').factory 'GroupForm', ->
+  templateUrl: 'generated/components/group_form/group_form.html'
+  controller: ($scope, $rootScope, $location, group, FormService, Records, $translate, PrivacyString) ->
+    $scope.group = group.clone()
+
+    $scope.$on 'modal.closing', (event) ->
+      FormService.confirmDiscardChanges(event, $scope.group)
+
+    $scope.i18n = do ->
+      h = {}
```

Comment:
Yea I agree. Can do. It's a big improvement over having conditions in the template though.